DS13/code/modules/overmap/ai.dm

```diff
@@ -0,0 +1,86 @@
+
+/obj/structure/overmap //AI vars
+	var/AI_enabled = FALSE //Do you want this ship to be computer controlled?
+	var/faction = "starfleet" //Change this if you want to make teams of AIs
+	var/obj/structure/overmap/target = null //Used for AIs, lets us track a target.
+	var/behaviour = "retaliate" //By default. Ais will only shoot back.
+	var/list/possible_behaviours = list("aggressive", "retaliate", "peaceful") //This will be admin selectable when I add an overmap panel
+	var/range = 15 //Firing range.
+
+/obj/structure/overmap/AI
+	name = "Romulan warbird class light cruiser"
+	desc = "A dangerous ship which resembles a bird. It has a modest armament and is highly maneuverable."
+	icon = 'DS13/icons/overmap/warbird.dmi'
+	icon_state = "warbird"
+	main_overmap = FALSE
+	class = "enemy-romulan" //Feel free to add overmap controls for AIs later, future me.
+	damage_states = FALSE //Damage FX
+	damage = 10 //Will turn into 20 assuming weapons powered
+	AI_enabled = TRUE //Start with an AI by default
+	faction = "romulan" //Placeholder
+
+/obj/structure/overmap/proc/take_control() //Take control of our ship, make it into an AI
+	START_PROCESSING(SSobj, src) //Need to process to check for targets and so on.
+	name = "[name] ([rand(0,1000)])"
+	shield_power = 2 //So theyre not ultra squishy
+	weapon_power = 1
+	engine_power = 1
+	max_shield_power = 4
+	max_weapon_power = 4
+	max_engine_power = 4
+	power_slots -= 4
+	AI_enabled = TRUE //Let the computer take the wheel
+	check_power()
+
+/obj/structure/overmap/proc/remove_control() //Stop the AI from controlling the ship
+	AI_enabled = FALSE
+	STOP_PROCESSING(SSobj,src)
+	name = initial(name)
+	nav_target = null
+	target = null
+
+/obj/structure/overmap/process()
+	if(!AI_enabled) //Process is ONLY called for ai ships. We don't want a non ai ship doing this or consequences
+		return
+	if(vel < max_speed)
+		vel += acceleration
+	if(!process) //Start process allows the ship to move. It is called at init but if for some reason it stops, we need to reboot it.
+		process = TRUE
+		start_process()
+	if(target) //We have a target locked in
+		if(get_dist(src, target) > range) //Target ran away. Move on.
+			target = null
+			nav_target = null
```

Comment:
I'd recommend keeping them as the nav_target, so the AI gives chase rather than letting you dash in and out of firing range.